internal/actions/uuid_module.go

```diff
@@ -0,0 +1,40 @@
+package actions
+
+import (
+	"context"
+
+	"github.com/dop251/goja"
+	"github.com/zitadel/logging"
+
+	"github.com/google/uuid"
+)
+
+func WithUUID(ctx context.Context) Option {
+	return func(c *runConfig) {
+		c.modules["zitadel/uuid"] = func(runtime *goja.Runtime, module *goja.Object) {
+			requireUUID(ctx, runtime, module)
+		}
+	}
+}
+
+func requireUUID(ctx context.Context, runtime *goja.Runtime, module *goja.Object) {
+	o := module.Get("exports").(*goja.Object)
+	logging.OnError(o.Set("v5", newSHA1(ctx, runtime))).Warn("unable to set module")
+}
+
+func newSHA1(ctx context.Context, runtime *goja.Runtime) func(call goja.FunctionCall) goja.Value {
+	return func(call goja.FunctionCall) goja.Value {
+		if len(call.Arguments) != 2 {
+			logging.WithFields("count", len(call.Arguments)).Debug("other than 2 args provided")
+			panic("invalid arg count")
+		}
+
+		space, err := uuid.Parse(call.Arguments[0].Export().(string))
+		if err != nil {
+			logging.WithError(err).Debug("space failed parsing as UUID")
+			panic(err)
+		}
+
+		return runtime.ToValue(uuid.NewSHA1(space, call.Arguments[1].Export().([]byte)).String())
+	}
+}
```

Comment:
it would be great if you could provide a small unit test for the logic of the `newSHA1` function

Reply:
ah BTW as the `ctx` is not used, i think you can omit it

Reply:
one less argument.
also i restructured it to remove some duplication with other versions, it now mirrors [the main js uuid package](https://www.npmjs.com/package/uuid)